### ISSUE TITLE
Arc not configured, preconfigure user/mail/author

### ISSUE THREAD

**Issue by rugk** [JanitorTechnology/janitor#432]
The trouble with the source code submission tools (ref https://github.com/JanitorTechnology/janitor/issues/423), continues…

Now _just before my patch was going to land_ someone noticed me that the diff does not have author info. **No tool** noticed me of this before!

> Diff does not have proper author information in Phabricator. See the Lando FAQ for help with this error. 

see https://wiki.mozilla.org/Phabricator/FAQ#Lando

---

So, I guess this is the issue:
> This can happen if you have not set an author email in your .hgrc file (or git config). Set your author email in your .hgrc to your username, Firstname Lastname <yourldapemail@mozilla.com>.

**Comment by jankeromnes** [comment]
@rugk Thanks a lot for reporting this issue!

I'm not quite sure what went wrong. In theory, both `.gitconfig` and `.hgrc` should be automatically generated for you.

You can check their values under https://janitor.technology/settings/ > Configurations.

Could you please verify that your name is correctly set up in your settings, and re-generate or fix these configuration files? (You can also re-deploy them to every container by using the link below the text area.)

**Comment by rugk** [comment]
Ah, yeah, this is how it was configured: https://vim.cx/?ede76a907e4e4683#xhQN8H/HfALnn3/+eU/lxkBoQkbfaKtQWTXz4xfRabo=

Thanks for the pointer, I could not adjust the files there, too, but I wonder why it did not initially configure it correctly.

**Comment by jankeromnes** [comment]
> but I wonder why it did not initially configure it correctly.

I suspect that's because Janitor can't know your real name when you sign in via email, so it leaves it blank unless you fill it manually.

When you sign in with GitHub (you can do that from your "Integrations" settings), it should automatically fill in your full name if it's not already set.

I think that enabling "Sign in with GitHub" generally for the Janitor web app could solve that problem.

**Comment by rugk** [comment]
I think, I signed in with the mail variant it uses only, then it generated the config files, and only then I adjusted the name in the Janitor page _and_ connected it with GitHub then.
It seems it did not trigger a config regeneration.

**Comment by jankeromnes** [comment]
> It seems it did not trigger a config regeneration.

Yes, we don't automatically re-generate config files, because users might already have customized them, and we don't want to overwrite valuable changes.

But maybe we could detect when config files are not customized, and overwrite them (or if they are customized, show a helpful warning to the user, and maybe a diff view between old a new config?)
